index.js

```diff
@@ -316,6 +316,10 @@ function HttpRequest(options) {
 			}
 		};
 	}
+
+	if ('onerror' in xhr && !xhr.onreadystatechange) {
+		xhr.onerror = onLoad;
```

Comment:
Can't we use `onreadystatechange` by default, as it's the more convenient? (Only one listener)
And support onload/onerror/ontimeout only for IE?